cluster_management/src/main/java/com/pinterest/rocksplicator/Utils.java

```diff
@@ -472,10 +477,55 @@ public static void checkStateTransitions(String fromState, String toState, Messa
     LOG.error("From " + fromState + " to " + toState + " for " + partitionName);
   }
 
+  public static String getMetaParentClusterLocation(String cluster) {
+    return "/metadata/" + cluster;
+  }
+
+  public static String getMetaParentResourceLocation(String cluster, String resourceName) {
+    return getMetaParentClusterLocation(cluster) + "/" + resourceName;
+  }
+
   public static String getMetaLocation(String cluster, String resourceName) {
-    return "/metadata/" + cluster + "/" + resourceName + "/resource_meta";
+    return  getMetaParentResourceLocation(cluster, resourceName) + "/resource_meta";
   }
 
+  public static String getMetaData(
+      final CuratorFramework zkClient,
+      final String cluster,
+      final String resourceName,
+      final int max_retries) throws Exception {
+    final String clusterMetaPath = Utils.getMetaParentClusterLocation(cluster);
+    final String resourceMetaPath = Utils.getMetaParentResourceLocation(cluster, resourceName);
+    final String metadataPath = Utils.getMetaLocation(cluster, resourceName);
+
+    final AtomicInteger retryCount = new AtomicInteger(0);
+    return RetryLoop.callWithRetry(zkClient.getZookeeperClient(), new Callable<String>() {
+      @Override
+      public String call() throws Exception {
+        retryCount.incrementAndGet();
+        try {
+          zkClient.sync().forPath(clusterMetaPath);
+          zkClient.sync().forPath(resourceMetaPath);
+          zkClient.sync().forPath(metadataPath);
+          return new String(zkClient.getData().forPath(metadataPath));
+        } catch (KeeperException.NoNodeException exp) {
+          LOG.error(String.format(
+              "MetaDataPath (%s) not visible or doesn't exist yet: ", metadataPath), exp);
+          // By throwing the OperationTimeoutException, we force the retry, as NoNode exception is
+          // not retryable.
```

Comment:
Is there any wiki pointing to what types of exe are retryable, and what types are not?

Reply:
Check RetryLoop.shouldRetry and ReteryLoop.isRetryException to understand what Exceptions are automatically retried by RetryLoop in the CuratorFramework.